tests/lib.rs

```diff
@@ -27,27 +27,75 @@ fn test_format() {
     let doc = doc! {
         "float" => 2.4,
         "string" => "hello",
-        "array" => ["testing", 1],
+        "array" => ["testing", 1, true, [1, 2]],
         "doc" => {
             "fish" => "in",
             "a" => "barrel",
-            "!" => 1
+            "!" => 1,
```

Comment:
Is the trailing comma required? If not, we should probably add a separate test for without it (both nested and top-level)